src/meta/processors/admin/ListClusterInfoProcessor.cpp

```diff
@@ -89,16 +105,27 @@ void ListClusterInfoProcessor::process(const cpp2::ListClusterInfoReq&) {
     hostServices[metaAddr.host].push_back(service);
   }
 
-  // check: there should be only one agent in each host
-  for (const auto& [host, services] : hostServices) {
+  // Check there is at least one agent in each host. If there are many ones, we will pick the first.
+  std::unordered_map<std::string, std::vector<cpp2::ServiceInfo>> agentServices;
+  for (auto& [host, services] : hostServices) {
     int agentCount = 0;
-    for (auto& s : services) {
-      if (s.get_role() == cpp2::HostRole::AGENT) {
+    for (auto it = services.begin(); it != services.end();) {
+      if (it->get_role() == cpp2::HostRole::AGENT) {
         agentCount++;
+
+        if (agentCount > 1) {
+          LOG(INFO) << folly::sformat(
+              "Will erase redundant agent {} in host {}", it->get_addr().toString(), host);
```

Comment:
erase from

Reply:
done~

Reply:
I mean "Will erase redundant agent {} from host {}", not important, up to you